presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainFilter.java

```diff
@@ -38,6 +38,8 @@
 
     boolean testNull();
 
+    boolean testNonNull();
```

Comment:
@highker In a general case `testNonNull()` can't be evaluated, e.g. it is not possible to tell whether a non-null value matches a filter or not without knowing what the value is. However, in case of `is null` and `is not null` filters, it is possible to tell whether non-value value passes or not. `testNonNull` exists to support `is [not] null` positional filters on nested columns, e.g. `a[1] is null`, where `a` is `array(array(T))`.